DOC-1693/create-a-how-to-guide-for-onboarding-migration (#627)
swan-io/docs: docs/topics/onboarding/overview/guide-migrate.mdx

Refer to the [individual](/topics/onboarding/individual/guide-create) and [company](/topics/onboarding/company/guide-create) creation guides for complete mutation examples.

### Step 3: Handle new required fields

The current API collects additional data for risk assessment and compliance.
Some fields that were optional or didn't exist in the previous API are now required.

Check the field reference pages for the complete list of requirements by country:

- [Individual onboarding fields](/topics/onboarding/individual/fields)
- [Company onboarding fields](/topics/onboarding/company/fields)

### Step 4: Update response handling

The current API returns typed responses instead of the generic `Onboarding` type.

- Individual mutations return `IndividualAccountHolderOnboarding`.
- Company mutations return `CompanyAccountHolderOnboarding`.
- The `finalizeAccountHolderOnboarding` mutation returns a `FinalizeAccountHolderOnboardingPayload`. On success, the payload contains a `FinalizeAccountHolderOnboardingSuccessPayload` with an `onboarding` field of type `AccountHolderOnboarding` (union).

Update your response parsing to handle the new type names and field structure.

> ### Step 5: Update registry data handling (company onboardings)

Author: ```suggestion
### Step 5: Update registry data handling (French company onboardings)
```

> If you onboard French companies and relied on automatic field population from official registries, you need to update your workflow.

Reviewer: Does this **only** apply to French company onboardings?

The current API does **not** auto-populate onboarding fields with registry data.

> Instead, call the `companyInfoRegistryData` query with the company's `registrationNumber` and `residencyAddressCountry`, then pass the results to the create or update mutation.

Author: ```suggestion
This step applies only if you onboard French companies. If you don't, skip to Step 6.

Previously, Swan auto-populated empty onboarding fields with data from the French National Business Register (RNE). The current API doesn't.

Instead, call the `companyInfoRegistryData` query with the company's `registrationNumber` and `residencyAddressCountry`, then pass the results to the create or update mutation.
```

### Step 6: Test in Sandbox

Validate your updated integration in Sandbox before going live.
Refer to the [testing checklist](#testing-checklist) for key scenarios.

### Step 7: Switch in Live

After successful Sandbox testing, deploy your changes to your Live environment.

## In-progress onboardings

> In-progress onboardings created with the previous API are not affected by the migration.

Reviewer: Could be made more direct: "The migration doesn't affect in-progress onboardings created with the previous API." But this is minor.

> In-progress onboardings created with the previous API are not affected by the migration.

Author: ```suggestion
The migration doesn't affect in-progress onboardings created with the previous API.
```

- Continue to update and finalize them using the previous mutations.
- They don't need to be recreated with the current API.
- Once finalized, any future onboardings should use the current API.

> After the previous mutations are removed at the end of September 2026, all in-progress onboardings using the previous API must already be finalized.

Reviewer: What happens if they're not?

> After the previous mutations are removed at the end of September 2026, all in-progress onboardings using the previous API must already be finalized.

Author: ```suggestion
After the previous mutations are removed at the end of September 2026, you can only update or finalize onboardings through the current API mutations, including any in-progress onboardings created under the previous API. Make sure your integration is fully migrated by then.
```

Plan your migration timeline to ensure no onboardings are left in an incomplete state.

## Deprecation timeline

| Date | Event |
|---|---|
| March 2026 | Current onboarding API available in beta. |
| April 2026 | Documentation updated. Previous mutations marked as deprecated across all guide pages. |

> | September 2026 | Previous mutations and queries removed. |

Reviewer: For consistency & accuracy, this should say "end of September 2026."

> | September 2026 | Previous mutations and queries removed. |

Author: ```suggestion
| End of September 2026 | Previous mutations and queries removed. |
```

## Testing checklist {#testing-checklist}

Before going live with the current API, validate the following scenarios in Sandbox:

1. Create an individual onboarding with all required fields for your target countries.
1. Create a company onboarding with `relatedIndividuals` (at least one legal representative and one UBO).
1. Update an existing onboarding to confirm nested input structure works correctly.
1. Finalize an onboarding and confirm the typed response (`IndividualAccountHolderOnboarding` or `CompanyAccountHolderOnboarding`).
1. For French companies: call `companyInfoRegistryData` and use the results to pre-fill the create mutation.
1. Test public onboarding mutations if you use public onboarding links.

> 1. Confirm webhook payloads still match your expectations for onboarding status changes.

Reviewer: Which webhook event types are we referring to here?

> 1. Confirm webhook payloads still match your expectations for onboarding status changes.

Author: ```suggestion
1. Confirm that [webhook payloads](/developers/using-api/webhooks) for `Onboarding.Created` and `Onboarding.Updated` still match your expectations.
```

Update your response parsing to handle the new type names and field structure.

### Step 5: Update registry data handling (French company onboardings)

This step applies only if you onboard French companies. If you don't, skip to Step 6.

Previously, Swan auto-populated empty onboarding fields with data from the French National Business Register (RNE). The current API doesn't.

Instead, call the `companyInfoRegistryData` query with the company's `registrationNumber` and `residencyAddressCountry`, then pass the results to the create or update mutation.

### Step 6: Test in Sandbox

Validate your updated integration in Sandbox before going live.
Refer to the [testing checklist](#testing-checklist) for key scenarios.

### Step 7: Switch in Live

After successful Sandbox testing, deploy your changes to your Live environment.

## In-progress onboardings

The migration doesn't affect in-progress onboardings created with the previous API.

- Continue to update and finalize them using the previous mutations.
- They don't need to be recreated with the current API.
- Once finalized, any future onboardings should use the current API.

> After the previous mutations are removed at the end of September 2026, you can only update or finalize onboardings through the current API mutations, including any in-progress onboardings created under the previous API. Make sure your integration is fully migrated by then.

Author: ```suggestion
After the previous mutations are removed at the end of September 2026, you'll no longer be able to update or finalize onboardings through them. Make sure your integration is fully migrated by then.
```
